File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int altitude = 0;
        int maxAltitude = 0;
        int len = gain.length;
        
        for (int i = 0; i < len; i++) {
            altitude += gain[i];
            maxAltitude = Math.max(maxAltitude, altitude);
        }
        
        return maxAltitude;        
    }
}

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.rs
impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let mut altitude = 0;
        let mut max_altitude = 0;
        
        for &g in &gain {
            altitude += g;
            max_altitude = std::cmp::max(max_altitude, altitude);
        }
        
        max_altitude
    }
}
